conda/base/context.py

```diff
@@ -792,16 +837,37 @@ def channels(self):
         if self._argparse_args and 'channel' in self._argparse_args:
             # TODO: it's args.channel right now, not channels
             argparse_channels = tuple(self._argparse_args['channel'] or ())
-            # Add condition to make sure that sure that we add the 'defaults'
+            # Add condition to make sure that we add the 'defaults'
             # channel only when no channels are defined in condarc
-            # We needs to get the config_files and then check that they
+            # We need to get the config_files and then check that they
             # don't define channels
             channel_in_config_files = any('channels' in context.raw_data[rc_file].keys()
                                           for rc_file in self.config_files)
             if argparse_channels and not channel_in_config_files:
                 return tuple(IndexedSet((*local_add, *argparse_channels, DEFAULTS_CHANNEL_NAME)))
 
-        return tuple(IndexedSet((*local_add, *self._channels)))
+        channels = self.channel_parameters.keys()
+
+        return tuple(IndexedSet((*local_add, *channels)))
+
+    @property
+    def channel_parameters(self) -> frozendict:
+        """
+        Return parameters associated with all channels if there are any.
+        It will be an empty frozendict if there are no parameters.
+        """
+        channel_params = {}
+
+        for channel in self._channels:
+            if isinstance(channel, Mapping):
+                keys = channel.keys()
+                if len(keys) > 0:
+                    name, *_ = keys
+                    channel_params[name] = channel.get(name)
+            else:
+                channel_params[channel] = frozendict()
```

Comment:
```suggestion
        for channel in filter(None, self._channels):
            if isinstance(channel, Mapping):
                name = next(iter(channel))
                channel_params[name] = channel.get(name)
            else:
                channel_params[channel] = frozendict()
```

Reply:
Nice usage of `filter`, `next` and `iter` 👍 🔥